lib/ramble/docs/results.rst

```diff
@@ -0,0 +1,86 @@
+.. Copyright 2022-2024 The Ramble Authors
+
+   Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+   https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+   <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+   option. This file may not be copied, modified, or distributed
+   except according to those terms.
+
+.. _workspace:
+
+================
+Ramble Results
+================
+
+Ramble provides support and tooling around processing results data.
+
+See ``ramble workspace analyze`` (and associated :doc:`workspace documentation<workspace>`) for generating such data.
+
+Most notably, the two supported functionalities are:
+  * Uploading of arbitrary JSON
+  * Generation of plots and analysis as part of a report
+
+-------------------
+Upload
+-------------------
+
+The `ramble results upload` command is used to imports experiment results from
+a JSON file, and uploads them as specified in the upload block of Ramble's
+config file.
+
+This can be used for both results from an arbitrary ramble workspace, or for
+data that is generated by tools which conform to rambles data schema.
+
+-------------------
+Report
+-------------------
+The `ramble results report` command is experimental functionality that provides
+the user a way to generate plots and reports from the output of a workspace.
+This output is in the form of individual `png` plots, and a combined `pdf`
+report
+
+Multiple plot types are supported, to aid in comparison of runs and common tasks, including:
+  * Scaling (line) plots (both strong and weak)
+  * Comparison bar plots
+  * General FOM bar plots
+
+For these plots, common operations are supported such as log axis, data
+grouping, and idealized line plotting.
+
+While this functionality is very useful and powerful, it is not intended as a
+wholesale replacement for uploading the data into more advanced offline
+analysis tools.
```

Comment:
I don't think this will ever totally be true? We will always want to support both CUJs? 

Or put another way this is never going to be so good we remove `--upload`?

Reply:
Definitely not, but I also think eventually it might be good enough that people who don't ever use `--upload` can do what they want with it.

For us, we always want `--upload` so we can have the results stored in a common place later.